packages/pds/src/api/app/bsky/feed/getPostThread.ts

```diff
@@ -53,169 +50,134 @@ export default function (server: Server, ctx: AppContext) {
   })
 }
 
-const getAncestors = async (
-  db: DatabaseSchema,
-  imgUriBuilder: ImageUriBuilder,
-  parentUri: string,
-  requester: string,
-): Promise<GetPostThread.Post | GetPostThread.NotFoundPost> => {
-  const parentRes = await postInfoBuilder(db, requester)
-    .where('post.uri', '=', parentUri)
-    .executeTakeFirst()
-  if (!parentRes) {
-    return {
-      $type: 'app.bsky.feed.getPostThread#notFoundPost',
-      uri: parentUri,
-      notFound: true,
+const composeThread = (
+  threadData: PostThread,
+  feedService: FeedService,
+  posts: PostInfoMap,
+  actors: ActorViewMap,
+  embeds: FeedEmbeds,
+) => {
+  const post = feedService.formatPostView(
+    threadData.post.postUri,
+    actors,
+    posts,
+    embeds,
+  )
+
+  let parent
+  if (threadData.parent) {
+    if (threadData.parent instanceof ParentNotFoundError) {
+      parent = {
+        $type: 'app.bsky.feed.getPostThread#notFoundPost',
```

Comment:
If forget, are there codegen constants kicking around for these that we can use?

Reply:
I don't think so. They haven't made it into the genned `ids` map yup. but we should have that in mind. i'll peel off an issue